kazu/steps/ner/tokenized_word_processor.py

```diff
@@ -224,81 +204,98 @@ def process_next_word(self, word: TokenizedWord) -> None:
-    """
+class MultilabelSpanFinder(SpanFinder):
+    """A span finder that can handle multiple labels per token, as opposed to the
+    standard 'BIO' format."""
 
-    def __init__(self, threshold: float, text: str, id2label: dict[int, str]):
+    def __init__(self, text: str, id2label: dict[int, str]):
+        """
+
+        :param text: the raw text to be processed
+        :param id2label: id to class label mappings
+        """
         super().__init__(text, id2label)
-        self.text = text
-        self.threshold = threshold
-        # we include a whitespace here so that new spans (potential entities) are started for each word
-        self.span_breaking_chars = {"(", ")", ";", " "}
+        self.active_labels: set[str] = set()
 
-    def get_bio_and_class_labels(self, word: TokenizedWord) -> set[tuple[str, Optional[str]]]:
-        """Returns bio and class labels if their confidence is above the configured
-        threshold.
+    def _update_active_spans(self, class_label: str, word: TokenizedWord) -> None:
+        """Updates any active spans.
 
-        :param word:
+        :param class_label: class label to update in active spans
+        :param word: TokenizedWord to append to active spans
+        """
+        for span in self.active_spans:
+            if class_label == span.clazz:
+                # the word is the same class as the active TokWordSpan, so extend it
+                span.tok_words.append(word)
+
+    def start_span(self, class_label: str, word: TokenizedWord) -> None:
+        """Start a new TokWordSpan for the given class label.
+
+        :param class_label: the label to use
+        :param word: the word to start the span with
         :return:
         """
-        bio_and_class_labels: set[tuple[str, Optional[str]]] = set()
-        indices_of_confs_above_threshold = torch.argwhere(word.token_confidences > self.threshold)
+        span = TokWordSpan(clazz=class_label, tok_words=[word])
+        self.active_spans.append(span)
+        self.active_labels.add(class_label)
+
+    def close_spans(self, class_label: str) -> None:
+        """Close any active spans."""
+        for active_span in list(self.active_spans):
+            if class_label == active_span.clazz:
+                self.closed_spans.append(active_span)
+                self.active_spans.remove(active_span)
+                self.active_labels.remove(class_label)
+
+    def __call__(self, words: list[TokenizedWord]) -> list[TokWordSpan]:
+        """Find spans of entities.
+
+        :param words:
+        :return: The spans found
+        """
+        for word in words:
+            self.process_next_word(word)
+        for clazz in set(self.active_labels):
+            self.close_spans(clazz)
+        return self.closed_spans
+
+    def get_class_labels(self, word: TokenizedWord) -> set[str]:
+        class_labels: set[str] = set()
+        indices_of_confs_above_threshold = torch.argwhere(word.token_confidences > 0)
```

Comment:
Hi, I just came across this line—does `word.token_confidences` represent logits? 
Previously, the threshold was 0.5, suggesting it was a probability (between 0 and 1). However, based on this commit (`torch.argwhere(word.token_confidences > 0)`), it seems it can have negative values, so I guess it represents logits in the current implementation.

It's been a while since I last looked at this, so I might have misunderstood the code. 😅 Please ignore if that's the case!

Reply:
Hi @wonjininfo ! Thanks for the check - actually, I've changed your implementation slightly - token confidences are now either 0 or 1. I should probably change these variable names as they're confusing

Reply:
Thanks for the explanation! These lines look good to me